clin/processor.py

```diff
@@ -83,6 +83,12 @@ def apply_sql_query(self, env: str, spec: dict):
         nakadi_sql = self._get_nakadi_sql(env)
         query = SqlQuery.from_spec(spec)
 
+        def is_allowed_sql_change(diff: dict):
+            for k in diff.get("values_changed", {}).keys():
+                if "auth" not in k:
+                    return False
```

Comment:
Help me understand why each changed value needs to have an `auth` section in order for the change to be valid for Nakadi SQL? In what way is such `auth` section specific to Nakadi SQL?

Reply:
On second thought, the logic here may actually not be accurate. I'm assuming that @Dmitry-Erokhin has a good reason for the change and has tested it. Unfortunately I'm not able to test myself since I don't have access Nakadi SQL...

But to explain the original logic: The only element that can be changed is the `auth` section. Since that's a list, it's never in `values_changed`, but only in `items_added` or `items_removed` (or something similar). That means if `diff` countains `values_changed` we can assume that it's a change to the output event type or the metatdata, which are both forbidden.

Reply:
Nakadi SQL allows updating only auth section. in this case, diff of changes can include 2 types of changes: 
- "iterable_added"/"iterable_removed" (not sure about exact naming): can happen only in auth identifying added/removed users/services to auth section
- "value_changed"
  - in auth section: change of any token value, hence allowed
  - not in auth section should be prohibited – it is changes of the query itself